### ISSUE TITLE
Suggestion: Variant of up and down which navigates directly to child or parent of current commit

### ISSUE THREAD

**Issue by mikebeaton** [ForkIssues/TrackerWin#1678]
Suggestion:

Somewhat related to my previous ticket #1676 (that it is sometimes hard to see what commit is the parent of what in the UI), but also I think extremely useful anyway, would be some variant of up and down (e.g. ALT+Up and ALT+Down?) which instead of navigating to the next newest or next oldest commit, navigates to the child (when there is (just) one, or beeps otherwise) and to the parent (there always is just one, in the HEAD~1 sense, except for any root commits) of the currently highlighted commit.

Apologies if I've missed the existence of such a feature already? I have had a quick look and not seen it.

Thank you.

**Comment by sublayer** [comment]
I agree, a way to navigate to a child commit would be immensely helpful in big projects with lots of branches. Sometimes, I will click the parent commit link, then not be able to quickly navigate back to the child commit without manually scrolling through the graph. Even a simple back/forward shortcut would help there, but a dedicated child link/shortcut would be ideal.

**Comment by mikebeaton** [comment]
Aha, there is a parent link, down in the detail window for the commit itself, so that goes one way - thank you @sublayer.

Obviously I'd still like to suggest adding a method of navigating the other way, to the child.

**Comment by DanPristupov** [comment]
What should we do when a commit has multiple children (a very common case):

<img width="209" alt="Screenshot 2023-02-02 at 9 25 36" src="https://user-images.githubusercontent.com/618115/216270719-ff7e1378-f196-41b1-97a0-2ce50f030405.png">

**Comment by mikebeaton** [comment]
On the currently selected (checked out) branch, I would say. Maybe when this is still not definitive (branch then subsequent merge) it can just beep/flash/refuse?

(EDIT: Or as in the parent link(s) case, it can also have multiple children links, down in that same place? Maybe as well as having an up and down key type control, with the above-mentioned rule? 🙏🙂)

**Comment by arphox** [comment]
> What should we do when a commit has multiple children (a very common case):
> 
> <img alt="Screenshot 2023-02-02 at 9 25 36" width="209" src="https://user-images.githubusercontent.com/618115/216270719-ff7e1378-f196-41b1-97a0-2ce50f030405.png">

I think this question is partially solved when we talk about parents, too.  
In case of parents, all parents are displayed as links, so this could work the same with child commits as well.

It could be here if it's not that hard (in performance) to get it fast (maybe a data structure for this could work):  
![kép](https://github.com/fork-dev/TrackerWin/assets/17320094/646f670c-7721-4fc3-beb8-725ad59ddae8)

**Comment by DanPristupov** [comment]
Related issue: https://github.com/fork-dev/TrackerWin/issues/845
